### PR TITLE
ENH: use /tmp/xoscar as default temp dir

### PR DESCRIPTION
**Pull request by UranusSeven**
## What do these changes do?
Use /tmp/xoscar as default temp dir

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: python/xoscar/backends/communication/socket.py [hunk 1]
```diff
@@ -258,12 +259,24 @@ async def connect(
...
-TEMPDIR = tempfile.gettempdir()
+def _get_or_create_default_unix_socket_dir():
+    import sys
+
+    if sys.platform.startswith("win"):
+        unix_socket_dir = XOSCAR_UNIX_SOCKET_DIR_WIN
+    else:
+        unix_socket_dir = XOSCAR_UNIX_SOCKET_DIR
+    os.makedirs(unix_socket_dir, exist_ok=True)
+    os.chmod(unix_socket_dir, mode=0o777)
```
Comment (qianduoduo0904) [inline python/xoscar/backends/communication/socket.py:270]: Is it possible that the user that runs the script does not have the permission to run this?

Comment (UranusSeven) [reply python/xoscar/backends/communication/socket.py:270]: Good point. How about catch the possible permission err and print a warning?

File: python/xoscar/backends/communication/socket.py [hunk 2]
```diff
@@ -258,12 +259,29 @@ async def connect(
         return SocketClient(local_address, dest_address, channel)
 
 
-TEMPDIR = tempfile.gettempdir()
+def _get_or_create_default_unix_socket_dir():
+    import sys
+
+    if sys.platform.startswith("win"):
+        unix_socket_dir = XOSCAR_UNIX_SOCKET_DIR_WIN
```
Comment (qianduoduo0904) [inline python/xoscar/backends/communication/socket.py:266]: Why does this line not covered by windows CI?

Comment (UranusSeven) [reply python/xoscar/backends/communication/socket.py:266]: This is unexpected. I'll double check it on a Windows PC.

**Comment by qianduoduo0904** [comment]
I am just curious that if it's possible to use the user directory, e.g. ~/xoscar ? and the permission error would not happen in this situation.

**Comment by UranusSeven** [comment]
> I am just curious that if it's possible to use the user directory, e.g. ~/xoscar ? and the permission error would not happen in this situation.

For a standalone deployment, i.e. an xoscar cluster starts from command line, `~/xoscar` is OK. But for an embedded deployment, like in Xorbits where user may start an Xorbits local cluster under any dir, you may want a fixed dir for temp files.

**Comment by qianduoduo0904** [comment]
> > I am just curious that if it's possible to use the user directory, e.g. ~/xoscar ? and the permission error would not happen in this situation.
> 
> For a standalone deployment, i.e. an xoscar cluster starts from command line, `~/xoscar` is OK. But for an embedded deployment, like in Xorbits where user may start an Xorbits local cluster under any dir, you may want a fixed dir for temp files.

Got it.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**